src/parser/parse_tree_cl.c

```diff
@@ -8352,7 +8353,8 @@ pt_print_datatype (PARSER_CONTEXT * parser, PT_NODE * p)
 
     case PT_TYPE_NUMERIC:
       q = pt_append_nulstring (parser, q, pt_show_type_enum (p->type_enum));
-      if (p->info.data_type.precision != 15 || p->info.data_type.dec_precision != 0)
+      if (p->info.data_type.precision != DB_DEFAULT_NUMERIC_PRECISION
+	  || p->info.data_type.dec_precision != DB_DEFAULT_NUMERIC_SCALE)
```

Comment:
Always true if DB_DEFAULT_NUMERIC_PRECISION and DB_DEFAULT_NUMERIC_SCALE are not equal.

Reply:
False only if NUMERIC (15, 0).

Reply:
All right. I was looking at the code wrong
You were comparing different variables with precision and dec_precision.
I mistakenly thought it was the same variable.

Reply:
It is confusing because it is not called scale here, but dec_precision.